app/src/main/java/fr/free/nrw/commons/upload/FileProcessor.java

```diff
@@ -81,6 +98,61 @@ GPSExtractor processFileCoordinates(SimilarImageInterface similarImageInterface)
         return imageObj;
     }
 
+    /**
+     * Redacts EXIF and XMP metadata as indicated in preferences.
+     *
+     */
+    @SuppressLint("CheckResult")
+    private void redactMetadata(Context context) {
```

Comment:
- Remove `@SuppressLint("CheckResult")`. 
- Dispose the `Observable`
- Write unit tests for the method
- Break down the method if possible